### PR TITLE
Add k8s dryrun test for galley

### PR DESCRIPTION
**Pull request by jasonwzm**
Please provide a description for what this PR is for.

fix https://github.com/istio/istio/issues/12803

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/galley/validation_test.go [hunk 1]
```diff
@@ -123,6 +123,16 @@ func TestValidation(t *testing.T) {
 							t.Fatalf("config request denied for wrong reason: %v", err)
 						}
 					}
+
+					wetRunErr := env.ApplyContents(ns.Name(), ym)
+					defer func() { _ = env.DeleteContents(ns.Name(), ym) }()
+
+					if err != nil && wetRunErr == nil {
```
Comment (howardjohn) [inline tests/integration/galley/validation_test.go]: is it even possible for `err != nil` at this point, due to the `t.Fatal` calls above?

Comment (jasonwzm) [reply tests/integration/galley/validation_test.go]: yes. if there is an error, the testdata is not valid and the error string is one of the recognized k8s api server errors, the error still exists. Here we want to test the dryrun and wetrun both return errors or neither return errors.

**Comment by jasonwzm** [comment]
@howardjohn PTL again when you have a chance. Thanks!

**Pull request merged**